src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs

```diff
@@ -124,7 +124,7 @@ private bool ProcessFieldsIfNeeded()
             {
                 var property = _csvReader.Context.Reader.HeaderRecord[i];
 
-                if (_csvReader.Context.Reader.HeaderRecord[i][0] == '@')
+                if (_csvReader.Context.Reader.HeaderRecord[i].StartsWith('@'))
```

Comment:
`_csvReader.Context.Reader.HeaderRecord[i]` cannot be null ?

Reply:
If CSV has a header, then all `HeaderRecords` entries will be not-null. 
If we have nothing between separator chars, we'll get `string.Empty` value.